File: QUEUE/QueueArray.cs
using System;

// COLA CON ARRAYS
class Program {
    const int MAXSIZE = 5;
    static int[] queue = new int[MAXSIZE];
    static int front = -1, rear = -1;
    
    static void Insertar() {
        Console.Write("Ingrese el elemento: ");
        string input = Console.ReadLine();
        int elem;
        
        if (!int.TryParse(input, out elem)) {
            Console.WriteLine("Entrada inválida");
            return;
        }
        
        if (rear == MAXSIZE - 1) {
            Console.WriteLine("OVERFLOW");
            return;
        }
        if (front == -1 && rear == -1) {
            front = rear = 0;
        } else {
            rear++;
        }
        queue[rear] = elem;
        Console.WriteLine("Elemento insertado correctamente");
    }
    
    static void Eliminar() {
        if (front == -1 || front > rear) {
            Console.WriteLine("UNDERFLOW");
            return;
        }
        int elemento = queue[front];
        if (front == rear) {
            front = rear = -1;
        } else {
            front++;
        }
        Console.WriteLine("Elemento eliminado " + elemento);
    }
    
    static void Mostrar() {
        if (rear == -1 || front == -1 || front > rear) {
            Console.WriteLine("La cola esta vacia");
        } else {
            Console.WriteLine("Elementos de la cola: ");
            for (int i = front; i <= rear; i++) {
                Console.WriteLine(queue[i]);
            }
        }
    }
    
    static void Main() {
        int opc = 0;
        
        while (opc != 4) {
            Console.WriteLine("\n************************MENU QUEUE****************************");
            Console.WriteLine("==================================================================");
            Console.WriteLine("1. Insertar elemento en la cola");
            Console.WriteLine("2. Eliminar elemento de la cola");
            Console.WriteLine("3. Mostrar la cola");
            Console.WriteLine("4. Salir");
            Console.Write("Ingrese su opcion: ");
            
            if (!int.TryParse(Console.ReadLine(), out opc)) {
                opc = 0;
            }
            
            switch(opc) {
                case 1:
                    Insertar();
                    break;
                case 2:
                    Eliminar();
                    break;
                case 3:
                    Mostrar();
                    break;
                case 4: 
                    Console.WriteLine("Saliendo del programa...");
                    break;
                default:
                    Console.WriteLine("\nOpcion invalida. Ingrese de nuevo\n");
                    break;
            }
        }
    }
}

File: QUEUE/QueueArray.py
# COLA CON ARRAYS
MAXSIZE = 5
queue = [0] * MAXSIZE
front = -1
rear = -1

def insertar():
    global front, rear
    elem = int(input("Ingrese el elemento: "))
    
    if rear == MAXSIZE - 1:
        print("OVERFLOW")
        return
    if front == -1 and rear == -1:
        front = rear = 0
    else:
        rear += 1
    queue[rear] = elem
    print("Elemento insertado correctamente")

def eliminar():
    global front, rear
    if front == -1 or front > rear:
        print("UNDERFLOW")
        return
    elemento = queue[front]
    if front == rear:
        front = rear = -1
    else:
        front += 1
    print(f"Elemento eliminado {elemento}")

def mostrar():
    if rear == -1 or front == -1 or front > rear:
        print("La cola esta vacia")
    else:
        print("Elementos de la cola: ")
        for i in range(front, rear + 1):
            print(queue[i])

def main():
    opc = 0
    
    while opc != 4:
        print("\n************************MENU QUEUE****************************")
        print("==================================================================")
        print("1. Insertar elemento en la cola")
        print("2. Eliminar elemento de la cola")
        print("3. Mostrar la cola")
        print("4. Salir")
        
        try:
            opc = int(input("Ingrese su opcion: "))
        except ValueError:
            opc = 0
        
        if opc == 1:
            insertar()
        elif opc == 2:
            eliminar()
        elif opc == 3:
            mostrar()
        elif opc == 4:
            print("Saliendo del programa...")
        else:
            print("\nOpcion invalida. Ingrese de nuevo\n")

if __name__ == "__main__":
    main()